src/peft/tuners/fourierft/layer.py

```diff
@@ -91,29 +96,9 @@ def get_delta_weight(self, adapter) -> torch.Tensor:
         indices = self.indices[adapter].to(spectrum.device)
         dense_spectrum = torch.zeros(self.out_features, self.in_features, device=spectrum.device)
         dense_spectrum[indices[0, :], indices[1, :]] = spectrum.float()
-        delta_weight = torch.fft.ifft2(dense_spectrum).real * self.fourierft_scaling[adapter]
+        delta_weight = torch.fft.ifft2(dense_spectrum, norm=self.kwargs['ifft2_norm']).real * self.fourierft_scaling[adapter]
```

Comment:
I think we need to make a change with regard to how we save `ifft2_norm`. Instead of keeping it inside of `kwargs`, in the `__init__`, we should create a dict `self.fourierft_ifft2_norm = {}` and in `update_layer`, we set: `self.fourierft_ifft2_norm[adapter_name] = ifft2_norm`. Then, instead of `self.kwargs['ifft2_norm']` here and below, always use `self.fourierft_ifft2_norm[adapter_name]`. This way, we can have multiple adapters with different values for this argument.

When doing this, don't forget to add `"fourierft_ifft2_norm"` to `other_param_names`.